src/sagemaker_training/smdataparallel.py

```diff
@@ -86,11 +86,17 @@ def _wait_for_workers(self):  # type: () -> None
         logger.info("Waiting for MPI workers to establish their SSH connections")
 
         workers = [host for host in self._hosts if host != self._master_hostname]
-        with timeout.timeout(seconds=self.timeout_in_seconds):
-            for host in workers:
-                while not _can_connect(host):
-                    time.sleep(self._interval)
-                logger.info("Worker %s available for communication", host)
+        try:
+            with timeout.timeout(seconds=self.timeout_in_seconds):
+                for host in workers:
+                    while not _can_connect(host):
+                        time.sleep(self._interval)
```

Comment:
On an average, how many attempts will this make to connect given the `timeout_in_seconds` interval` and `_interval` ?

Reply:
The `self._interval` time is 1 sec and in general default timeout is 3600 secs

Reply:
Thanks.